iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -803,12 +818,34 @@ private Schema schema(Properties properties, org.apache.hadoop.hive.metastore.ap
       // Add partitioning columns to the original column list before creating the Iceberg Schema
       List<FieldSchema> cols = Lists.newArrayList(hmsTable.getSd().getCols());
       cols.addAll(hmsTable.getPartitionKeys());
-      return HiveSchemaUtil.convert(cols, autoConversion);
+      Schema schema = HiveSchemaUtil.convert(cols, autoConversion);
+      return getSchemaWithIdentifierFields(schema, identifierFields);
```

Comment:
what is the diff between those 2 if blocks? can't we combine them
````
if (hmsTable.isSetPartitionKeys() {
     cols.addAll(hmsTable.getPartitionKeys());
}
````
rest is copy-paste

Reply:
Simplified this code block.